From: Jiri Slaby <jslaby@suse.cz>
Date: Mon, 22 May 2017 11:20:47 +0200
Subject: Revert "wlcore: Add RX_BA_WIN_SIZE_CHANGE_EVENT event"
Patch-mainline: never, kabi
References: kabi

This reverts commit d4392269f7ce3567e3bd7c2b43883433c4d64a37, upstream
commit e7ee74b56f23ba447d3124f2eccc32033cca501d. It depends on commit
8ef67e0078b30ed1b4c36671723d585f80ab903c, upstream commit
480dd46b9d6812e5fb7172c305ee0f1154c26eed. But that is going to be
reverted due to kABI breakage.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/net/wireless/ti/wl18xx/event.c | 28 ----------------------------
 drivers/net/wireless/ti/wl18xx/event.h |  1 -
 drivers/net/wireless/ti/wl18xx/main.c  |  3 +--
 3 files changed, 1 insertion(+), 31 deletions(-)

diff --git a/drivers/net/wireless/ti/wl18xx/event.c b/drivers/net/wireless/ti/wl18xx/event.c
index 085ef5c87262..09c7e098f460 100644
--- a/drivers/net/wireless/ti/wl18xx/event.c
+++ b/drivers/net/wireless/ti/wl18xx/event.c
@@ -206,33 +206,5 @@ int wl18xx_process_mailbox_events(struct wl1271 *wl)
 						 mbox->sc_pwd_len,
 						 mbox->sc_pwd);
 
-	if (vector & RX_BA_WIN_SIZE_CHANGE_EVENT_ID) {
-		struct wl12xx_vif *wlvif;
-		struct ieee80211_vif *vif;
-		struct ieee80211_sta *sta;
-		u8 link_id = mbox->rx_ba_link_id;
-		u8 win_size = mbox->rx_ba_win_size;
-		const u8 *addr;
-
-		wlvif = wl->links[link_id].wlvif;
-		vif = wl12xx_wlvif_to_vif(wlvif);
-
-		/* Update RX aggregation window size and call
-		 * MAC routine to stop active RX aggregations for this link
-		 */
-		if (wlvif->bss_type != BSS_TYPE_AP_BSS)
-			addr = vif->bss_conf.bssid;
-		else
-			addr = wl->links[link_id].addr;
-
-		sta = ieee80211_find_sta(vif, addr);
-		if (sta) {
-			sta->max_rx_aggregation_subframes = win_size;
-			ieee80211_stop_rx_ba_session(vif,
-						wl->links[link_id].ba_bitmap,
-						addr);
-		}
-	}
-
 	return 0;
 }
diff --git a/drivers/net/wireless/ti/wl18xx/event.h b/drivers/net/wireless/ti/wl18xx/event.h
index 9495fadc8093..f3d4f13379cb 100644
--- a/drivers/net/wireless/ti/wl18xx/event.h
+++ b/drivers/net/wireless/ti/wl18xx/event.h
@@ -38,7 +38,6 @@ enum {
 	REMAIN_ON_CHANNEL_COMPLETE_EVENT_ID      = BIT(18),
 	DFS_CHANNELS_CONFIG_COMPLETE_EVENT       = BIT(19),
 	PERIODIC_SCAN_REPORT_EVENT_ID            = BIT(20),
-	RX_BA_WIN_SIZE_CHANGE_EVENT_ID           = BIT(21),
 	SMART_CONFIG_SYNC_EVENT_ID               = BIT(22),
 	SMART_CONFIG_DECODE_EVENT_ID             = BIT(23),
 	TIME_SYNC_EVENT_ID                       = BIT(24),
diff --git a/drivers/net/wireless/ti/wl18xx/main.c b/drivers/net/wireless/ti/wl18xx/main.c
index 47f355e92193..50cce42089a5 100644
--- a/drivers/net/wireless/ti/wl18xx/main.c
+++ b/drivers/net/wireless/ti/wl18xx/main.c
@@ -1029,8 +1029,7 @@ static int wl18xx_boot(struct wl1271 *wl)
 		DFS_CHANNELS_CONFIG_COMPLETE_EVENT |
 		SMART_CONFIG_SYNC_EVENT_ID |
 		SMART_CONFIG_DECODE_EVENT_ID |
-		TIME_SYNC_EVENT_ID |
-		RX_BA_WIN_SIZE_CHANGE_EVENT_ID;
+		TIME_SYNC_EVENT_ID;
 
 	wl->ap_event_mask = MAX_TX_FAILURE_EVENT_ID;
 
-- 
2.13.0

